contracts/modules/SwapsExternal.sol

```diff
@@ -91,4 +98,18 @@ contract SwapsExternal is VaultController, SwapsUser {
     ) external view returns (uint256) {
         return _swapsExpectedReturn(sourceToken, destToken, sourceTokenAmount);
     }
+
+    function checkPriceDivergence(        
+        address sourceToken,
+        address destToken,
+        uint256 sourceTokenAmount,
+        address priceFeeds_) 
+        public 
+        returns(bool result) 
+    {
+        (uint256 rate, uint256 precision) = IPriceFeeds(priceFeeds_).queryRate(sourceToken, destToken);
+        uint256 destTokenAmount = _swapsExpectedReturn(sourceToken, destToken, sourceTokenAmount);
+        uint256 rateFromSwap = destTokenAmount.mul(precision).div(sourceTokenAmount);
+        result = rate == rateFromSwap ? true : false;
```

Comment:
just `result = rate == rateFromSwap;`
using `true` and `false` in comparisons and other logical expressions is generally discouraged